example/js2/api/ICU4XFixedDecimalFormatterOptions.mjs

```diff
@@ -0,0 +1,63 @@
+// generated by diplomat-tool
+
+import { ICU4XFixedDecimalGroupingStrategy } from "./ICU4XFixedDecimalGroupingStrategy.mjs"
+import wasm from "./diplomat-wasm.mjs";
+import * as diplomatRuntime from "./diplomat-runtime.mjs";
+
+
+
+export class ICU4XFixedDecimalFormatterOptions {
+    #groupingStrategy;
+    get groupingStrategy()  {
+        return this.#groupingStrategy;
+    }
+    set groupingStrategy(value) {
+        this.#groupingStrategy = value;
+    }
+    #someOtherConfig;
+    get someOtherConfig()  {
+        return this.#someOtherConfig;
+    }
+    set someOtherConfig(value) {
+        this.#someOtherConfig = value;
+    }
+
+    // Return this struct in FFI function friendly format.
+    // Returns an array that can be expanded with spread syntax (...)
+    
+    _intoFFI() {
+        return [
+            groupingStrategy.ffiValue, 
+            someOtherConfig]
```

Comment:
Observation: this turns a struct into an ordered list of its fields, which is then spread into separate arguments when passed into the WASM export. Is that correctly aligned with what WASM expects?

Reply:
IIRC yes? but yes, let's double check that

Reply:
I just did, through the LLVM IR output:

```ll
; diplomat_example::decimal::ffi::ICU4XFixedDecimalFormatter::try_new
; Function Attrs: nounwind
define hidden align 4 ptr @_ZN16diplomat_example7decimal3ffi26ICU4XFixedDecimalFormatter7try_new17h1c21e7448442dd44E
(ptr align 4 %locale, ptr align 4 %provider, i32 %options.0, i1 zeroext %options.1)
unnamed_addr #0 !dbg !7776 {
```

`i32 %options.0` is most likely `grouping_strategy` and `i1 %options.1` is most likely `some_other_config`.